ArduPlane/AP_Arming.cpp

```diff
@@ -334,6 +335,7 @@ void AP_Arming_Plane::update_soft_armed()
 
     hal.util->set_soft_armed(_armed);
     AP::logger().set_vehicle_armed(hal.util->get_soft_armed());
+    AP::logger().set_vehicle_powered(AP::battery().powered());
```

Comment:
Depending on how the argument goes (usb vs battery), just have the Logger code talk to the battery code directly.  You don't need the state that way.

The block of code you've dropped stuff into here isn't something I like - I'd rather not have the current `set_vehicle_armed` but we play silly games so we can get arming check results in on failed arming.